prov/psm2/src/psmx2_util.c

```diff
@@ -87,7 +87,7 @@ char *psmx2_uuid_to_string(psm2_uuid_t uuid)
 		uuid[4], uuid[5], uuid[6], uuid[7], uuid[8], uuid[9],
 		uuid[10], uuid[11], uuid[12], uuid[13], uuid[14], uuid[15]);
 
-	return s;
+	return strdup(s);
```

Comment:
I anticipate that static analysis tools will complain that strdup is not checked for success (maybe at the calling site?), and that the string is not freed if successful.

Reply:
The newly updated PR moves strdup() to the call site where the allocation is needed (and doesn't expect to be freed before the process terminates).